test/integration/viewer_spec.mjs

```diff
@@ -454,4 +454,354 @@ describe("PDF viewer", () => {
       );
     });
   });
+
+  describe("Detail view on zoom", () => {
+    const BASE_MAX_CANVAS_PIXELS = 1e6;
+
+    function setupPages(zoom, devicePixelRatio) {
+      let pages;
+
+      beforeAll(async () => {
+        pages = await loadAndWait(
+          "colors.pdf",
+          null,
+          zoom,
+          {
+            // When running Firefox with Puppeteer, setting the
+            // devicePixelRatio Puppeteer option does not properly set
+            // the `window.devicePixelRatio` value. Set it manually.
+            earlySetup: `() => {
+              window.devicePixelRatio = ${devicePixelRatio};
+            }`,
+          },
+          { maxCanvasPixels: BASE_MAX_CANVAS_PIXELS * devicePixelRatio ** 2 },
+          { height: 600, width: 800, devicePixelRatio }
+        );
+      });
+
+      afterAll(async () => {
+        await closePages(pages);
+      });
+
+      return function forEachPage(fn) {
+        return Promise.all(
+          pages.map(([browserName, page]) => fn(browserName, page))
+        );
+      };
+    }
+
+    function extractCanvases(pageNumber) {
+      const pageOne = document.querySelector(
+        `.page[data-page-number='${pageNumber}']`
+      );
+      return Array.from(pageOne.querySelectorAll("canvas"), canvas => {
+        const { width, height } = canvas;
+        const ctx = canvas.getContext("2d");
+        const topLeft = ctx.getImageData(2, 2, 1, 1).data;
+        const bottomRight = ctx.getImageData(width - 3, height - 3, 1, 1).data;
+        return {
+          size: width * height,
+          topLeft: globalThis.pdfjsLib.Util.makeHexColor(...topLeft),
+          bottomRight: globalThis.pdfjsLib.Util.makeHexColor(...bottomRight),
+        };
+      });
+    }
+
+    function waitForDetailRendered(page) {
+      return createPromise(page, resolve => {
+        const controller = new AbortController();
+        window.PDFViewerApplication.eventBus.on(
+          "pagerendered",
+          ({ source }) => {
+            if (source.constructor.name === "PDFPageDetailView") {
```

Comment:
Here, and elsewhere, having to rely explicitly on the constructor name feels a little "brittle" to me.

How about instead adding a `isDetailView` property to the "pagerendered" event dispatching (i.e. similar to the existing `cssTransform`)?

Reply:
That would mean some "public API" for something that we only use in tests. Are you ok with that?

Reply:
Actually yes, this is a good idea as consumers that are currently waiting for a page to be render will in many cases want to add an `!isDetailView` check.

Reply:
I was thinking that it'd just be a boolean value (assuming that works for the tests), so we'd not expose any actual functionality that way.

Reply:
> Actually yes, this is a good idea as consumers that are currently waiting for a page to be render will in many cases want to add an `!isDetailView` check.

Yes, and we may want to add a `|| evt.isDetailView` check to the early return here:
https://github.com/mozilla/pdf.js/blob/d1f62509e531b9e9f8862e3526eaa17565ff1e9c/web/pdf_viewer.js#L867-L870